packages/agents-hosting/src/oauth/userTokenClient.ts

```diff
@@ -123,4 +123,50 @@ export class UserTokenClient {
       return { status: TokenRequestStatus.Failed, token: undefined }
     }
   }
+
+  /**
+   * Gets the token or sign-in resource.
+   * @param userId The user ID.
+   * @param connectionName The connection name.
+   * @param channelId The channel ID.
+   * @param conversation The conversation reference.
+   * @param relatesTo The related conversation reference.
+   * @param code The code.
+   * @param finalRedirect The final redirect URL.
+   * @param fwdUrl The forward URL.
+   * @returns A promise that resolves to the token or sign-in resource response.
+   */
+  async getTokenOrSignInResource (userId: string, connectionName: string, channelId: string, conversation: ConversationReference, relatesTo: ConversationReference, code: string, finalRedirect: string = '', fwdUrl: string = '') : Promise<TokenOrSinginResourceResponse> {
+    const state = Buffer.from(JSON.stringify({ conversation, relatesTo, connectionName, msAppId: this.msAppId })).toString('base64')
+    const params = { userId, connectionName, channelId, state, code, finalRedirect, fwdUrl }
+    const response = await this.client.get('/api/usertoken/GetTokenOrSignInResource', { params })
+    return response.data as TokenOrSinginResourceResponse
+  }
+
+  /**
+   * Gets the token status.
+   * @param userId The user ID.
+   * @param channelId The channel ID.
+   * @param include The optional include parameter.
+   * @returns A promise that resolves to the token status.
+   */
+  async getTokenStatus (userId: string, channelId: string, include: string = null!): Promise<TokenStatus[]> {
+    const params = { userId, channelId, include }
+    const response = await this.client.get('/api/usertoken/GetTokenStatus', { params })
+    return response.data as TokenStatus[]
+  }
+
+  /**
+   * Gets the AAD tokens.
+   * @param userId The user ID.
+   * @param connectionName The connection name.
+   * @param channelId The channel ID.
+   * @param resourceUrls The resource URLs.
+   * @returns A promise that resolves to the AAD tokens.
+   */
+  async getAadTokens (userId: string, connectionName: string, channelId: string, resourceUrls: AadResourceUrls) : Promise<Record<string, TokenResponse>> {
```

Comment:
I dont know how to validate with different `resourceUrls` the param is not documented, or used in dotnet.